Sources/Appcast/SUAppcastDriver.swift

```diff
@@ -8,16 +8,85 @@
+            // If skipped major version is >= than the item's minimumAutoupdateVersion, we can skip the item.
+            // But if there is an ignoreSkippedUpgradesBelowVersion, we can only skip the item if the last skipped subrelease
+            // version is >= than that version provided by the item
+            return true
+        }
+
+        if let skippedMinorVersion = skippedUpdate.minorVersion, versionComparator.compareVersion(skippedMinorVersion, toVersion: item.versionString) != .orderedAscending {
+            // Item is on a less or equal version than a minor version we've skipped
+            // So we skip this item
+            return true
+        }
+
+        return false
+    }
+    
+    public static func filterSupportedAppcast(_ appcast: SUAppcast, phasedUpdateGroup: NSNumber?, skippedUpdate: SPUSkippedUpdate?, currentDate: Date, hostVersion: String, versionComparator: SUVersionComparison, testOSVersion: Bool, testMinimumAutoupdateVersion: Bool) -> SUAppcast {
+
+        let hostPassesSkippedMajorVersion = SPUAppcastItemStateResolver.isMinimumAutoupdateVersionOK(skippedUpdate?.majorVersion, hostVersion: hostVersion, versionComparator: versionComparator)
+
+        let filteredItems = appcast.items.filter { item in
+            let passesOSVersion = !testOSVersion || (item.minimumOperatingSystemVersionIsOK && item.maximumOperatingSystemVersionIsOK)
+            let passesPhasedRollout = itemIsReadyForPhasedRollout(item, phasedUpdateGroup: phasedUpdateGroup, currentDate: currentDate, hostVersion: hostVersion, versionComparator: versionComparator)
+            let passesMinimumAutoupdateVersion = !testMinimumAutoupdateVersion || !item.isMajorUpgrade
+            let passesSkippedUpdates = hostVersion.isEmpty || !containsSkippedUpdate(item: item, skippedUpdate: skippedUpdate, hostPassesSkippedMajorVersion: hostPassesSkippedMajorVersion, versionComparator: versionComparator)
+
+            return passesOSVersion && passesPhasedRollout && passesMinimumAutoupdateVersion && passesSkippedUpdates
+        }
+
+        return SUAppcast(items: filteredItems)
+    }
+    
+    public static func deltaUpdate(from appcastItem: SUAppcastItem, hostVersion: String) -> SUAppcastItem? {
+        return appcastItem.deltaUpdates[hostVersion]
     }
     
     public static func bestItem(fromAppcastItems: [SUAppcastItem], getDeltaItem: SUAppcastItem?, withHostVersion: String, comparator: SUVersionComparison) -> SUAppcastItem {
-        return fromAppcastItems.first!
+        var item: SUAppcastItem?
+        for appcastItem in fromAppcastItems {
+            if item == nil || comparator.compareVersion(item!.versionString, toVersion: appcastItem.versionString) == .orderedAscending {
+                item = appcastItem
+            }
+        }
+        return item!
```

Comment:
We cannot force unwrap `item!` here, as the filter may not find any suitable `SUAppcationItem` objects.

The return type of this method should be `SUAppcationItem?`.